core/src/main/scala/pickling/package.scala

```diff
@@ -0,0 +1,26 @@
+package scala
+
+package object pickling {
+
+  /** A combination of a static pickler and an unpickler for type `T`.
+   */
+  type SPicklerUnpickler[T] = SPickler[T] with Unpickler[T]
+  object SPicklerUnpickler {
+    import scala.language.experimental.macros
+
+    // these return the two traits rather than the alias just in case
+    // the alias would somehow have a different meaning in some context
+
+    def apply[T](p: SPickler[T], u: Unpickler[T]): SPickler[T] with Unpickler[T] = new SPicklerUnpicklerImpl(p, u)
```

Comment:
This really makes me think the alias thing is odd, if we need an implementation of the class anyway